homeassistant/components/simplisafe/__init__.py

```diff
@@ -523,27 +484,33 @@ async def update_system(system):
         for result in results:
             if isinstance(result, InvalidCredentialsError):
                 if self._emergency_refresh_token_used:
-                    _LOGGER.error(
-                        "SimpliSafe authentication disconnected. Please restart HASS"
+                    LOGGER.error(
+                        "Authentication disconnected or invalid. Please re-add the "
+                        "SimpliSafe integration in HASS"
```

Comment:
In the future we can start a reauth config flow step here. See eg the Samsung TV integration.

Reply:
Per our discussion below, will look to implement this here. 👍

Reply:
@MartinHjelmare Another "oof" with this one.

I don't store the user's SimpliSafe password in the config entry – I only store the refresh token for security purposes; you need the password to re-init MFA. I could update the config flow to store the password, but that would still require the user to delete their existing configuration and re-add (which doesn't help our immediate use). Unless there's a way to prompt a user (via notification) to enter new config info on an existing entry?

I'm going to remove the 0.113.1 tag for now; this deserves some thought.

Reply:
We'll start a new config flow via the reauth step. We can ask the user for whatever we need to reauthenticate. At the end we update the existing config entry and abort the flow.

Reply:
Roger, I'll give it a shot.